CMakeLists.txt

```diff
@@ -26,3 +26,24 @@ zephyr_library_sources(src/regression_6000.c)
 # ######################################################################################################################
 
 zephyr_library_include_directories(${ZEPHYR_OPTEELIB_MODULE_DIR}/public)
+
+# ######################################################################################################################
+# Prebuilt TAs
+# ######################################################################################################################
+
+if(TA_DEPLOY_DIR)
+    set(TA_BIN_DIRECTORY ta_bin)
+    set(GENERATED_DIRECTORY ta_table)
+    execute_process(
```

Comment:
I am not sure I got how it is supposed to work. Main goal of a build system is to track changes in the source files and rebuild all dependent files. In this case, CMake will not track if any of the TAs were rebuilt and will not regenerate `ta_table.c`. Unless I missed something, of course

Reply:
This way doesn't need enumerate necessary TAs in yaml and CMake. Also it simplified CMakeLists.txt. Regeneration and compilation `ta_table.*` files not a big deal and not significantly affects to overall build time.

Reply:
AFAIK, it would be done only during the first build, when you call `cmake` to create `Makefile`/`build.ninja`. If I understand how CMake works, all the following builds will not regenerate `ta_table.*` files:

> execute_process runs commands while CMake is configuring the project, prior to build system generation. Use the [add_custom_target()](https://cmake.org/cmake/help/latest/command/add_custom_target.html#command:add_custom_target) and [add_custom_command()](https://cmake.org/cmake/help/latest/command/add_custom_command.html#command:add_custom_command) commands to create custom commands that run at build time

(https://cmake.org/cmake/help/latest/command/execute_process.html)

Reply:
Just did some tests by calling
`west build -b rcar_h3ulcb_ca57 -p always -- -DTA_DEPLOY_DIR=$(pwd)/prebuilt`
several times. And I can confirm that ta_table is regenerating each time.

Reply:
> west build -b rcar_h3ulcb_ca57 -p always -- -DTA_DEPLOY_DIR=$(pwd)/prebuilt

Because `-p always` triggers full rebuild every time.

Reply:
fixed